src/bci_build/package/__init__.py

```diff
@@ -1019,26 +1019,64 @@ def readme_url(self) -> str:
 
     @property
     def readme(self) -> str:
+        """The contents of the image README that is refered by ``io.artifacthub.package.readme-url`` label.
+
+        The text content is obteined from :py:attr:`BaseContainerImage.extra_files` if
+        ``README.md`` or ``README.md.j2`` are defined.
+
+        If no files are explicitly defined in :py:attr:`BaseContainerImage.extra_files`
+        but they are available in the image folder, assuming :py:attr:`BaseContainerImage.name`
+        as the folder name, they are going to be used.
+
+        ``README.md`` will be copied as is while ``README.md.j2`` will be compiled
+        before being returned. ``README.md`` takes precedence over ``README.md.j2``.
```

Comment:
I know this just became apparent due to you writing the docs, so feel free to ignore it, but imho this behavior is confusing. If there's a README.md and a README.md.j2, then we should error out. I cannot imagine a situation where having both is **not** a bug.

Reply:
How it works is:

- Try README.md from `extra_files`
- Try README.md.j2 from `extra_files`
- Try README.md from the image folder
- Try README.md.j2 from the image folder

The search stops once one of these matches. If there are two files, the second is simply ignored.

I'm not sure what is the best option here.

In theory, it could raise an error if out of the 4 items, there is more than one that matches, because it is not clear which README should be used. In practice, it seems unlikely that we have both, but we don't test that.

Reply:
imho we should not have README.md from extra files going forward. it's just in transition as we had things in flight. going forward we should have only one way to specify READMEs. either as a verbatim file or as a .j2 variant in the package subdir.

Reply:
Now that we have a license partial template that should be used, I don't see a reason for a verbatim file, everything going forward seems to be a template, and even non-jinja files could be parsed as templates with no issues.

Or there are too many cases where Jinja would not be used?

Reply:
No, you're right, I think only supporting `README.md.j2` is enough

Reply:
I updated the code to accept only `README.md.j2` and for that, I also had to update a few old READMEs that used `README.md` as their name.